docs/src/main/asciidoc/undertow-reference.adoc

```diff
@@ -143,3 +143,27 @@ quarkus.http.cors.exposed-headers=Content-Disposition
 If you want to use the `keycloak` Quarkus extension in your project, you should configure CORS using the `keycloak` extension
 configuration. See link:keycloak-guide#configuring-cors[the Keycloak guide] for more details.
 ====
+
+== Server Additional Options
+It is possible to provide some options that control server behaviour. The following properties are supported.
+[cols="<m,<m,<2",options="header"]
+|===
+|Property Name|Default|Description
+|quarkus.http.server-options.max-headers|`200`| The maximum number of headers that are permitted in a request.
+|quarkus.http.server-options.max-parameters|`1000`|The maximum number of query parameters that are permitted in a request.
+|quarkus.http.server-options.max-header-size|`50K`|The maximum size of a HTTP header block.
+|quarkus.http.server-options.max-buffered-request-size|`16384`|The maximum size of a request that can be saved.
+|quarkus.http.server-options.max-entity-size| `unlimited` |The default maximum size of a request entity.
+|quarkus.http.server-options.multipart-max-entity-size| `unlimited` | The default max entity size when using the Multipart parser.
```

Comment:
As they are all related to `limits`, maybe I would have made them `quarkus.http.limits.max-entity-size`.

Just a suggestion though, wait for other opinions before changing.

Reply:
I have been thinking about this, I am sold to the idea. So let's do this.